### PR TITLE
packages/core: bit of auth refactoring and added StaticAuthSessionManager and AuthSessionStore

### PR DESCRIPTION
**Pull request by Rugvip**
Some reusable parts that'll be needed for GHE auth, maybe GitHub auth, and any other auth flow that doesn't refresh.

### CODE REVIEW

File: packages/core/src/api/apis/implementations/lib/AuthSessionManager/RefreshingAuthSessionManager.ts [hunk 1]
```diff
@@ -14,47 +14,40 @@
...
+export class RefreshingAuthSessionManager<T> implements SessionManager<T> {
+  private readonly connector: AuthConnector<T>;
+  private readonly helper: SessionScopeHelper<T>;
+  private readonly sessionScopesFunc: SessionScopesFunc<T>;
+  private readonly sessionShouldRefreshFunc: SessionShouldRefreshFunc<T>;
 
-  private refreshPromise?: Promise<AuthSession>;
-  private currentSession: AuthSession | undefined;
+  private refreshPromise?: Promise<T>;
+  private currentSession: T | undefined;
```
Comment (freben) [inline packages/core/src/api/apis/implementations/lib/AuthSessionManager/RefreshingAuthSessionManager.ts:48]: Mixing use of  `?` and `| undefined` :)

**Review by freben — APPROVED** [review]
Good stuff.

I'm lazy - what's actually stored? Can there be sensitive information ending up in local storage?

File: packages/core/src/api/apis/implementations/lib/AuthSessionManager/RefreshingAuthSessionManager.ts (hunk 1 above)
Comment (Rugvip) [reply packages/core/src/api/apis/implementations/lib/AuthSessionManager/RefreshingAuthSessionManager.ts:48]: yeah stuff copied from different places, cba x)

**Comment by Rugvip** [comment]
Depends on what's in the auth session, the storage should only be used with things that are fine to put in local storage. The docs mentions that it uses local storage so leaving the rest to the auth implementation.

**Pull request merged**